### PR TITLE
feat(extensions): support core24 in the gnome extensions

### PR DESCRIPTION
**Pull request by sergiusens**
- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `tox run -m lint`?
- [ ] Have you successfully run `tox run -e test-py310`? (supported versions: `py39`, `py310`, `py311`, `py312`)

-----

### CODE REVIEW

File: tests/spread/core24/linters-pack/library-gnome/snap/snapcraft.yaml [hunk 1]
```diff
@@ -74,7 +64,9 @@ parts:
       - -Ddisable-introspection=true
     organize:
       snap/gnome-calculator/current/usr: usr
-    build-snaps: [ gnome-42-2204-sdk/latest/edge ]
+    build-snaps:
+      - gnome-46-2404-sdk/latest/edge
+      - gnome-46-2404/latest/edge
```
Comment (kenvandine) [inline tests/spread/core24/linters-pack/library-gnome/snap/snapcraft.yaml:69]: is adding the content snap here for linting purposes?

Comment (sergiusens) [reply tests/spread/core24/linters-pack/library-gnome/snap/snapcraft.yaml:69]: yes, but no, we require it as it is defined in `default-provider` so we try to download the snap for it for linting purposes, this should go away once the snap is in stable

File: snapcraft/extensions/gnome.py [hunk 2]
```diff
@@ -24,8 +24,8 @@
 
 from .extension import Extension, get_extensions_data_dir, prepend_to_env
 
-_SDK_SNAP = {"core22": "gnome-42-2204-sdk"}
-_PLATFORM_TRANSLATION = {"core22": "2204"}
+_SDK_SNAP = {"core22": "gnome-42-2204-sdk", "core24": "gnome-46-2404"}
```
Comment (tigarmo) [inline snapcraft/extensions/gnome.py:27]: how come this is `"gnome-46-2404"` and not `"gnome-46-2404-sdk"?

Comment (sergiusens) [reply snapcraft/extensions/gnome.py:27]: good catch!

**Review by tigarmo — APPROVED** [review]

File: tests/spread/core24/linters-pack/library-gnome/snap/snapcraft.yaml [hunk 3]
```diff
@@ -44,16 +44,6 @@ apps:
     common-id: org.gnome.Calculator.desktop
 
 parts:
-  meson-deps:
```
Comment (lengau) [inline tests/spread/core24/linters-pack/library-gnome/snap/snapcraft.yaml:47]: 😃

**Review by lengau — APPROVED** [review]

**Review by syu-w — APPROVED** [review]

**Review by cmatsuoka — APPROVED** [review]
Looks good!

**Pull request merged**